acir/src/circuit/opcodes/black_box_function_call.rs

```
@@ -7,138 +7,376 @@ use serde::{Deserialize, Serialize};
+                let output = outputs[0];
+                BlackBoxFuncCall::AND { lhs, rhs, output }
+            }
+            BlackBoxFunc::XOR => {
+                let lhs = inputs[0];
+                let rhs = inputs[1];
+                let output = outputs[0];
+                BlackBoxFuncCall::XOR { lhs, rhs, output }
+            }
+            BlackBoxFunc::RANGE => {
+                let input = inputs[0];
+                BlackBoxFuncCall::RANGE { input }
+            }
+            BlackBoxFunc::SHA256 => BlackBoxFuncCall::SHA256 { inputs, outputs },
+            BlackBoxFunc::Blake2s => BlackBoxFuncCall::Blake2s { inputs, outputs },
+            BlackBoxFunc::ComputeMerkleRoot => BlackBoxFuncCall::ComputeMerkleRoot {
+                leaf: inputs[0],
+                index: inputs[1],
+                hash_path: inputs[2..].to_vec(),
+                output: outputs[0],
+            },
+            BlackBoxFunc::SchnorrVerify => BlackBoxFuncCall::SchnorrVerify {
+                public_key_x: inputs[0],
+                public_key_y: inputs[1],
+                signature: inputs[2..66].to_vec(),
+                message: inputs[66..].to_vec(),
+                output: outputs[0],
+            },
+            BlackBoxFunc::Pedersen => BlackBoxFuncCall::Pedersen { inputs, outputs },
+            BlackBoxFunc::HashToField128Security => {
+                BlackBoxFuncCall::HashToField128Security { inputs, output: outputs[0] }
+            }
+            BlackBoxFunc::EcdsaSecp256k1 => BlackBoxFuncCall::EcdsaSecp256k1 {
+                public_key_x: inputs[0..32].to_vec(),
+                public_key_y: inputs[32..64].to_vec(),
+                signature: inputs[64..128].to_vec(),
+                message: inputs[128..].to_vec(),
+                output: outputs[0],
+            },
+            BlackBoxFunc::FixedBaseScalarMul => {
+                BlackBoxFuncCall::FixedBaseScalarMul { input: inputs[0], outputs }
+            }
```

These will panic rather than returning an error if the ACIR is malformed. Can you change this to return an `InvalidData` error?